### PR TITLE
Research podman kube and openshift tests

### PR DESCRIPTION
**Pull request by majamassarini**
Fixes packit/deployment#539

### CODE REVIEW

File: research/testing/openshift-zuul-test-alternatives.md [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+E               kubernetes.config.config_exception.ConfigException: Invalid kube-config file. No configuration found.
+
+/usr/local/lib/python3.9/site-packages/kubernetes/config/kube_config.py:751: ConfigException
+```
+
+## Summary
+
+[`podman kube play`](https://docs.podman.io/en/v4.4/markdown/podman-kube-play.1.html) can not be used to test:
+
+- _sandcastle_; we need a k8s cluster to be able to use the `kubernates` library. We could deploy pods in the cluster using [`podman kube apply`](https://docs.podman.io/en/v4.4/markdown/podman-kube-apply.1.html) but still we need an up and running cluster.
```
Comment (TomasTomecek) [inline research/testing/openshift-zuul-test-alternatives.md:352]: we could also implement 'container' plugin for sandcastle though that would be an overkill solution

File: research/testing/openshift-zuul-test-alternatives.md [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+- _deployment_; the Openshift tests in the _deployment_ repo are checking that pods are up and running on an Openshift dev instance; we can not check the same using `podman kube play` or `podman kube apply` (we would test different deployment settings...).
+
+`podman kube play` can be used for openshift tests in _packit-service_ project not related with _sandcastle_ but `docker-compose` should be enough for these as well; so I don't really see advantages in using `podman kube play`.
+
+For tests in _deployment_, _sandcastle_ and in _packit-service_ (which reference _sandcastle_) we still need a running k8s cluster.
+
+If I get it correctly, the **strimzi** project has tests running on Testing Farm using _minikube_: https://developers.redhat.com/articles/2023/08/17/how-testing-farm-makes-testing-your-upstream-project-easier#
+For this reason I think we can probably make something similar using Openshift (maybe using Openshift Local - I think it makes sense to test everything against an Openshift instance).
```
Comment (TomasTomecek) [inline research/testing/openshift-zuul-test-alternatives.md:360]: +1

**Review by TomasTomecek — APPROVED** [review]
Thank you for the insightful research!

File: research/testing/openshift-zuul-test-alternatives.md (hunk 1 above)
Comment (majamassarini) [reply research/testing/openshift-zuul-test-alternatives.md:352]: I don't think I know what you mean 😅 but I believe you 🙂

Comment (TomasTomecek) [reply research/testing/openshift-zuul-test-alternatives.md:352]: Sorry, that wasn't fully correct what I wrote. In packit we have 2 command runners right now: local (via subprocess) or openshift (sandcastle).

We could create a 3rd one: execute a command inside a container (ideally in the sandcastle container image).

**Pull request merged**